src/pages/Checkout/index.tsx

```diff
@@ -0,0 +1,171 @@
+import { Box, Button, Center, Checkbox, Flex, FormControl, FormLabel, Image, Input, Select, Spinner, Text, useToast } from '@chakra-ui/react';
+import axios from 'axios';
+import React, { useEffect, useState } from 'react';
+import { useLocation, useNavigate, useParams } from 'react-router-dom';
+
+import { authSessionStorage } from '@/utils/storage';
+
+interface Product {
+  id: string;
+  name: string;
+  imageURL: string;
+  price: { sellingPrice: number };
+}
+
+const CheckoutPage: React.FC = () => {
+  const { productId } = useParams<{ productId: string }>();
+  const { state } = useLocation() as { state: { totalPrice: number; product: Product; quantity: number } };
+  const [product, setProduct] = useState<Product | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+  const [receiptInfo, setReceiptInfo] = useState({
+    applyReceipt: false,
+    receiptType: 'personal',
+    receiptNumber: '',
+  });
+  const toast = useToast();
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (state && state.product) {
+      setProduct(state.product);
+      setIsLoading(false);
+    } else {
+      const fetchProductDetail = async () => {
+        try {
+          const response = await axios.get(`https://kakao-tech-campus-mock-server.vercel.app/api/v1/products/${productId}/detail`);
+          setProduct(response.data.detail);
+          setIsLoading(false);
+        } catch (error) {
+          toast({
+            title: 'Error',
+            description: 'Product not found. Redirecting to home page.',
+            status: 'error',
+            duration: 2000,
+            isClosable: true,
+          });
+          navigate('/');
+        }
+      };
+      fetchProductDetail();
+    }
```

Comment:
이부분 if ~ try까지 뎁스가 좀 긴데, if-return패턴으로 분기하면 어떨까요

```suggestion
if (state & state.product) {
  setProduct(state.product);
  setIsLoading(false);
  return;
}

(async () => {...}();
```

Reply:
말씀해주신 내용이 훨씬 코드의 가독성을 높이는 방향인 것 같습니다, 분기했습니다 !!